src/main/java/org/mockito/internal/stubbing/answers/DefaultValidableAnswer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+
+package org.mockito.internal.stubbing.answers;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import static org.mockito.internal.exceptions.Reporter.wrongTypeReturnedByDefaultAnswer;
+
+public class DefaultValidableAnswer {
```

Comment:
Problem with the proposal is that I need the `invocation`, for type checking against the method signature and to raise an error if needed. Also it extract too much in the `MockHandlerImpl`